libs/filabridge/src/SamplerBindingMap.cpp

```diff
@@ -43,14 +44,23 @@ void SamplerBindingMap::populate(const SamplerInterfaceBlock* perMaterialSib,
         if (sib) {
             auto sibFields = sib->getSamplerInfoList();
             for (const auto& sInfo : sibFields) {
-                if (offset > maxSamplerIndex) {
+                if (numSampler > maxSamplerIndex) {
                     overflow = true;
                 }
                 addSampler({
                     .blockIndex = blockIndex,
                     .localOffset = sInfo.offset,
                     .globalOffset = offset++,
                 });
+                // Shader stage flags is only needed to check if MAX_SAMPLER_COUNT is exceeded.
+                // Somehow if we can get shader stage flags from Program then we can remove it in SamplerInterfaceBlock.
+                const auto stageFlags = sib->getStageFlags();
+                if (stageFlags.vertex) {
```

Comment:
I'm confused with this logic. As I understand it, we're still limited to 16 samplers _per stage_, correct? Wouldn't the following not trigger an overflow warning, but be invalid (because we're using 17 fragment samplers)?

```
  0 shadowMap { fragment }
  1 froxels { fragment }
  2 iblDFG { fragment }
  3 iblSpecular { fragment }
  4 ssao { fragment }
  5 ssr { fragment }
  6 structure { fragment }
  7 targets { vertex }
  8 test0 { vertex | fragment }
  9 test1 { vertex | fragment }
  10 test2 { vertex | fragment }
  11 test3 { vertex | fragment }
  12 test4 { vertex | fragment }
  13 test5 { vertex | fragment }
  14 test6 { vertex | fragment }
  15 test7 { vertex | fragment }
  16 test8 { vertex | fragment }
  17 test9 { vertex | fragment }
```

I think we also need to compare against `MAX_VERTEX_SAMPLER_COUNT` / `MAX_FRAGMENT_SAMPLER_COUNT` here.

Reply:
Yes it's a but. It's fixed.